pydicom/valuerep.py

```diff
@@ -43,6 +43,58 @@
 # (the component separator '=' is handled separately)
 PN_DELIMS = {0x5e}
 
+FIXED_VALUE_VRS = ("AS", "DA")
+
+MAX_VALUE_LEN = {
+    "AE": 16,
+    "AS": 4,
+    "CS": 16,
+    "DA": 8,
```

Comment:
DA is 18

Reply:
That's quite a lot of typos even for me...

Reply:
But only in the context of a query, otherwise is 8 ([Table 6.2-1](https://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html#table_6.2-1))

Reply:
Ah right - I actually took the values from that table, though sometimes incorrectly... In this specific case I thought that this is not a relevant case, but in pynetdicom it probably is. I also have been thinking about an option to switch off the checks completely - it depends on the performance impact they may have.

Reply:
I don't know how fussed I am about supporting the query lengths. As long as I can customise the checks then I can monkey patch from within pynetdicom. Not sure how many people would be working on query datasets without having pynetdicom.

Reply:
For now, I allow both, if needed we can make this configurable.